apps/framework-cli/src/framework/typescript/templates.rs

```diff
@@ -92,27 +89,31 @@ GROUP BY toStartOfDay(timestamp)
 
 export default {
   teardown: [
-    dropAggregation(MATERIALIZED_VIEW),
-    dropTable(DESTINATION_TABLE),
+    ...dropAggregation({
+      viewName: MATERIALIZED_VIEW,
+      tableName: DESTINATION_TABLE,
+    }),
   ],
   setup: [
-    createTable({
-      name: DESTINATION_TABLE,
-      columns: {
-        date: "Date",
-        dailyActiveUsers: "AggregateFunction(uniq, String)"
+    ...createAggregation({
+      tableCreateOptions: {
+        name: DESTINATION_TABLE,
+        columns: {
+          date: "Date",
+          dailyActiveUsers: "AggregateFunction(uniq, String)",
+        },
+        engine: ClickHouseEngines.AggregatingMergeTree,
+        orderBy: "date",
+      },
+      materializedViewCreateOptions: {
+        name: MATERIALIZED_VIEW,
+        destinationTable: DESTINATION_TABLE,
+        select: SELECT_QUERY,
+      },
+      populateTableOptions: {
+        destinationTable: DESTINATION_TABLE,
+        select: SELECT_QUERY,
```

Comment:
Looks like there is some repetition - for that `createAggregation` function could we consolidate some of those so that the user doesn't have to repeat?